api/router/exec.go

```diff
@@ -80,3 +80,80 @@ func (e *ExecRouter) ExecCreateHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	render.JSON(w, r, exec)
 }
+
+func (e *ExecRouter) ExecGetHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log.Ctx(ctx).Info().Msgf("Executing ExecGet request")
+
+	execID := chi.URLParam(r, "exec")
+	if execID == "" {
+		err := fmt.Errorf("missing execID")
+		render.Render(w, r.WithContext(ctx), types.ErrHTTPBadRequest(err, "Invalid request"))
+		return
+	}
+
+	exec, err := e.store.Get(execID)
+	if err != nil {
+		render.Render(w, r.WithContext(ctx), types.ErrHTTPError(err, "Failed to get session"))
+		return
+	}
+	render.JSON(w, r, exec)
+}
```

Comment:
so i think even if we have these handlers method calls to the exec store should go through the exec service, otherwise we will end up having business logic spread across the exec router and the exec get handler, and business logic inside of the 'router' and inside of the exec service

as an example consider the list handler, sometimes we want to list terminated execs with the response (i.e. in a special case of unweave ls where users also tell us they want terminated execs) and sometimes we don't (i.e. in the case of listing execs for ssh), however as below the only option we have is to pass listTerminated to the store directly, so this has been muddied.

i think as a temporary stop gap we can encapsulate an exec service in the exec router which doesn't have any driver and make sure each method in the exec service that requires a driver throws an error if a driver is not allocated to it

Reply:
Yeah that's my PR! Hence the comment in the description ;) This is a temp solution to get the overall workflow done.